### PR TITLE
docs: add passport adapter

### PR DESCRIPTION
**Pull request by jannyHou**
connect to https://github.com/strongloop/loopback-next/issues/3183

Add  "creating and registering passport based strategies" in our doc.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Loopback-component-authentication.md [hunk 1]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
 }
 ```
 
+### Use passport based strategies
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: > Use passport based strategies 

I believe the standard is that section titles have to be in uppercase

Using Passport-based Strategies

File: docs/site/Loopback-component-authentication.md [hunk 2]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
...
+
+The earlier version of `@loopback/authentication` is based on a express
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: > is based on a express

is based on an express

File: docs/site/Loopback-component-authentication.md [hunk 3]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
...
+middleware called `passport`, which supports 500+ passport strategies for
+verifying express app's requests. In `@loopback/authentication@2.0`, we defined
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: > verifying express app's requests. 

verifying an express app's requests.

File: docs/site/Loopback-component-authentication.md [hunk 4]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
...
+our own interface `AuthenticationStrategy` that describes a strategy with
+different contracts than the passport strategy, but we still want to keep the
+ability to support those existing 500+ community passport strategies. Therefore
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: >  Therefore 

 Therefore,

File: docs/site/Loopback-component-authentication.md [hunk 5]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
...
+we rewrote the adapter class which convert a passport strategy to the one that
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: > we rewrote the adapter class which convert a passport strategy to the one that

we rewrote the adapter class. It now converts a passport strategy to the one that

File: docs/site/Loopback-component-authentication.md (hunk 4 above)
Comment (b-admike) [inline docs/site/Loopback-component-authentication.md]: typo: `Therefore ` -> `Therefore, `

File: docs/site/Loopback-component-authentication.md [hunk 6]
```diff
@@ -608,6 +608,23 @@ export class MyApplication extends BootMixin(
...
+LoopBack 4 authentication system wants and released it in a new package
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: > LoopBack 4 authentication system wants and released it in a new package 

LoopBack 4 authentication system expects. The updated adapter class was released in a new package named

File: docs/site/Loopback-component-authentication.md [hunk 7]
```diff
@@ -801,7 +818,7 @@ overrides them with any request-level overriding options specified in the
 
 Then, in the `authenticate()` function of the custom authentication strategy,
 call the `processOptions()` function, and have the custom authentication
-strategy react to the upated options.
+strategy react to the updated options.
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: good catch, thx

File: docs/site/Loopback-component-authentication.md [hunk 8]
```diff
@@ -384,7 +384,7 @@ export class MyAuthenticatingSequence implements SequenceHandler {
 }
 ```
 
-Notice the new dependency injection in the sequence's contructor.
+Notice the new dependency injection in the sequence's constructor.
```
Comment (emonddr) [inline docs/site/Loopback-component-authentication.md]: good catch, thx

**Pull request merged**